src/release.ss

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env gxi-script
+;; -*- Gerbil -*-
+;;; Release script for Gerbil.
+;; Usage: ./release.ss [version]
+;;
+;; Use it from a fully committed git tree.
+;;
+;; If a release version is specified (contains no "-"), a new release commit is made and tagged.
+;; If a development version is specified, a new development commit is made and tagged.
+;;
+;; If no version is specified, one is autodetected:
+;; If the current commit is a development commit, a release commit will be made and tagged
+;; If the current commit is a release commit, a development commit will be made and tagged
+;; e.g. v0.12-DEV-666-g1234567 will be turned into v0.13
+;; and v0.13 will be turned into v0.13-DEV
+
+(import
+  :gerbil/gambit/ports
+  :std/format :std/misc/ports :std/srfi/13 :std/srfi/14 :std/sugar :std/pregexp)
+
+;; Where the root to gerbil is.
+(def gerbil-root
+  (path-normalize (path-expand ".." (path-directory (this-source-file)))))
+
+;; Change the directory to the root of gerbil.
+(current-directory gerbil-root)
+
+;; Check that .git is there indeed
+(unless (file-exists? ".git") (error "Cannot find .git directory"))
+
+;; Path to the version file
+(def version-path
+  "src/gerbil/runtime/gx-version.scm")
+
+;; Path to the .gitignore file
+(def ignore-path
+  ".gitignore")
+
+;; Error
+(def (raise-process-error status settings)
+  (error "Process exited with non-zero status" status settings))
+
+(def (check-process-error process settings)
+  (let ((status (process-status process)))
+    (unless (zero? status)
+      (raise-process-error status settings))))
+
+;; Get the output of a process, by default as a string.
+;; The caller may specify via the reader: argument an alternate function taking an input-port;
+;; the value returned will be that of the function.
+;; If the check-status?: argument is true, then the process exit status will be checked,
+;; and an error will be raised if it isn't 0.
+;; If the check-status?: argument is false,
+;; the process exit status will be returned as a second value.
+(def (process-output
+      path: path
+      arguments: (arguments '())
+      directory: (directory #f)
+      environment: (environment #f)
+      reader: (reader read-all-as-string)
+      check-status?: (check-status? #t)
+      show-console: (show-console #f)
+      stdin-redirection: (stdin-redirection #t)
+      stdout-redirection: (stdout-redirection #t)
+      stderr-redirection: (stderr-redirection #f))
+  (let* ((settings
+          [path: path
+           arguments: arguments
+           directory: directory
+           environment: environment
+           show-console: #f
+           stdin-redirection: #t
+           stdout-redirection: #t
+           stderr-redirection: #f])
```

Comment:
uhm shouldn't the settings be set from the variables? 
This blanket sets the settings to their defaults and ignores the keyword arguments.